app/frontend/src/features/messages/groupchats/groupChatsSlice.ts

```diff
@@ -0,0 +1,88 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import {
+  fetchGroupChatsThunk,
+  fetchMessagesThunk,
+  leaveGroupChatThunk,
+  sendMessageThunk,
+} from ".";
+import { GroupChat, Message } from "../../../pb/conversations_pb";
+
+const initialState = {
+  groupChats: [] as GroupChat.AsObject[],
+  error: "",
+  loading: false,
+  groupChatView: {
+    groupChat: null as GroupChat.AsObject | null,
+    messages: [] as Message.AsObject[],
+    error: "",
+    loading: false,
+  },
+};
+
+export type RequestsState = typeof initialState;
+
+export const groupChatsSlice = createSlice({
+  name: "groupChats",
+  initialState,
+  reducers: {
+    groupChatsFetched(state, action: PayloadAction<GroupChat.AsObject[]>) {
+      state.groupChats = action.payload;
+    },
+    setGroupChat(state, action: PayloadAction<GroupChat.AsObject | null>) {
+      state.groupChatView.groupChat = action.payload;
+    },
+    setMessages(state, action: PayloadAction<Message.AsObject[]>) {
+      state.groupChatView.messages = action.payload;
+    },
```

Comment:
I think it's better to remove this and set the messages in the  `extraReducers`

Reply:
I will check how that might look like.

Reply:
[This is why I mention it](https://redux.js.org/style-guide/style-guide#model-actions-as-events-not-setters)

Reply:
That does not tell why it should go to the `extraReducers`.

Reply:
I wasn't very clear, it's more that setting messages shouldn't be an action itself, it's part of fetching messages.